xonsh/ptk_shell/completer.py

```diff
@@ -114,6 +114,8 @@ def get_completions(self, document, complete_event):
                 yield comp
             else:
                 disp = comp[pre:].strip("'\"")
+                if not plen:
```

Comment:
does this plen is `0` or `None` here?

Reply:
This catches any False-y value. I think the only possibility is 0, since it ultimately comes from `xonsh.completer.complete_from_context()`, so we could compare with it directly.

Othewise, in case `plen==0` is a "valid" value, we could set it within the `elif len(completions) == 0:` clause in L74, which guarantees no "normal" completions were generated.

Reply:
`0` is definitely a valid value -- if you do `cd <TAB>` then every directory at your location will be returned and the prefix length will be `0` for all of them

Reply:
amended to change `plen` only if no regular completions found